services/src/main/java/org/keycloak/broker/provider/HardcodedGroupMapper.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.broker.provider;
+
+import org.jboss.logging.Logger;
+import org.keycloak.broker.oidc.mappers.AbstractClaimToGroupMapper;
+import org.keycloak.models.IdentityProviderMapperModel;
+import org.keycloak.models.IdentityProviderSyncMode;
+import org.keycloak.provider.ProviderConfigProperty;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author <a href="mailto:dmartino@redhat.com">Daniele Martinoli</a>
+ * @version $Revision: 1 $
+ */
+public class HardcodedGroupMapper extends AbstractClaimToGroupMapper {
+    protected static final List<ProviderConfigProperty> configProperties = new ArrayList<>();
+
+    private static final Logger LOG = Logger.getLogger(HardcodedGroupMapper.class);
+
+    private static final Set<IdentityProviderSyncMode> IDENTITY_PROVIDER_SYNC_MODES =
+            new HashSet<>(Arrays.asList(IdentityProviderSyncMode.values()));
+
+    static {
+        ProviderConfigProperty property;
+        property = new ProviderConfigProperty();
+        property.setName(ConfigConstants.GROUP);
+        property.setLabel("Group");
+        property.setHelpText("Group to assign the user.");
+        property.setType(ProviderConfigProperty.GROUP_TYPE);
+        configProperties.add(property);
```

Comment:
Could you please lazily create the properties list from the `getConfigProperties` method?

I understand we have other places using this pattern to initialize configuration properties but we are trying to avoid doing it. The reason is that it increases memory footprint by creating instances that are eventually used at runtime.